bundles/io/org.eclipse.smarthome.io.net.security/src/main/java/org/eclipse/smarthome/io/net/security/internal/ServiceConfiguration.java

```diff
@@ -0,0 +1,265 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.net.security.internal;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.Base64.Encoder;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Configuration of the {@link NetworkServerTlsProvider}.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class ServiceConfiguration {
+    // Multi-context data, needs to be parsed
+    String domains = "";
+    String psks = "";
+    String keystorePasswords = "";
+    String privateKeyPasswords = "";
+
+    // Self-signed certificate configuration data
+    public String organizationalUnit = "Eclipse Smarthome";
+    public String organization = "Eclipse Foundation, Inc.";
+    public String city = "Ottawa";
+    public String state = "Ontario";
+    public String country = Locale.getDefault().getCountry();
+    public BigDecimal validityInDays = BigDecimal.valueOf(365 * 3);
+
+    // Algorithm specific for new RSA or EC public/key pair
+    public String algorithmTLS = "RSA";
+    public String algorithmDTLS = "EC";
+    public String algorithmSigningTLS = "SHA1WithRSA";
+    public String algorithmSigningDTLS = "SHA256withECDSA";
+    public BigDecimal keysizeTLS = BigDecimal.valueOf(2048);
+    public BigDecimal keysizeDTLS = BigDecimal.valueOf(256);
+
+    // Management configuration
+    public Boolean autoRefreshInvalidCertificate = false;
+
+    // Parsed configuration contexts
+    protected Map<String, ContextConfiguration> contextConfigurations = new HashMap<>();
+    private File baseDir;
+
+    /**
+     * Parse the configuration fields, that contain multiple context-depending values and create
+     * corresponding {@link ContextConfiguration} objects.
+     *
+     * @param messageDigest A message digest for creating hash values over keystore files.
+     * @param baseDir The ESH home directory, used as base directory for determining keystore file paths.
+     */
+    public void parseContexts(MessageDigest messageDigest, File baseDir) {
+        this.baseDir = baseDir;
+        contextConfigurations.clear();
+
+        final Charset charSet = StandardCharsets.ISO_8859_1;
```

Comment:
Because of the rotten old properties standard of java. I need to write the configuration back to the configuration cfg file, and that needs to be ISO_8859_1 (because it was defacto standard 20 years ago)